### PR TITLE
feat: dropdown component

### PR DESCRIPTION
**Pull request by jessywork**
This update addresses the need for a customizable and reusable set of a dropdown component in the project.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/components/dropdown/src/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    borderColor: theme.colors.neutral.silver.shade70,
+    borderColorHover: theme.colors.neutral.silver.shade50,
+    borderColorActive: theme.colors.neutral.silver.shade30,
+    borderRadius: '0.25rem',
+    fontSize: theme.fontSize,
+    fontWeight: theme.fontWeight.REGULAR,
+    fontSizeLabel: theme.fontSize,
+    fontWeightLabel: theme.fontWeight.BOLD,
+    fontWeightHover: theme.fontWeight.BOLDER,
+    width: '30vw',
```
Comment (joaoprcosta) [inline packages/components/dropdown/src/Dropdown.tsx:50]: Be careful to use 'vw'. its better to use 'rem'

File: packages/components/dropdown/src/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      );
+      if (newSelectedItem) {
+        setSelectedItem(newSelectedItem);
+      }
+    } else {
+      setSelectedItem(undefined);
+    }
+  }, [selectedId, options]);
+
+  const handleChange = (item: DropdownItem) => {
```
Comment (joaoprcosta) [inline packages/components/dropdown/src/Dropdown.tsx:68]: Add 'useCallback' to memoize

File: packages/components/dropdown/src/dropdown.styles.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  &:active {
+    border-color: ${({ styled }) => styled.borderColorActive};
+  }
+
+  span {
+    transform: rotate(-90deg);
+  }
+
+  @media (max-width: ${({ theme }) => theme.breakpoints.tablet}px) {
+    width: 70vw !important;
```
Comment (joaoprcosta) [inline packages/components/dropdown/src/dropdown.styles.ts:56]: Never use !important

File: packages/components/dropdown/src/dropdown.styles.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  {
+    styled: Required<DropdownStyledProps>;
+    isOpen: boolean;
+  } & DropdownProps
+>`
+  list-style: none;
+  margin: 0;
+  padding: 0.5rem 0.75rem;
+  position: absolute;
+  z-index: 100;
```
Comment (joaoprcosta) [inline packages/components/dropdown/src/dropdown.styles.ts:70]: This value could be too high

**Review by JuniorDiasOliveira — APPROVED** [review]
Congrats!

**Review by Arthurmqz — APPROVED** [review]

**Pull request merged**